src/engine/TextLimit.cpp

```diff
@@ -0,0 +1,162 @@
+//  Copyright 2024, University of Freiburg,
+//                  Chair of Algorithms and Data Structures.
+//  Author: Nick Göckel <nick.goeckel@students.uni-freiburg.de>
+
+#include "./TextLimit.h"
+
+// _____________________________________________________________________________
+TextLimit::TextLimit(QueryExecutionContext* qec, const size_t n,
+                     std::shared_ptr<QueryExecutionTree> child,
+                     const ColumnIndex& textRecordColumn,
+                     const vector<ColumnIndex>& entityColumns,
+                     const vector<ColumnIndex>& scoreColumns)
+    : Operation(qec),
+      qec_(qec),
```

Comment:
No need to store the `qec_` in the textlimit, you can just retrieve it from the base class if required.